examples/create_upload_tutorial.py

```diff
@@ -77,8 +89,98 @@
+                       '+Wisconsin+(Diagnostic)'),
+    paper_url=('https://www.spiedigitallibrary.org/conference-proceedings-of'
+               '-spie/1905/0000/Nuclear-feature-extraction-for-breast-tumor-'
+               'diagnosis/10.1117/12.148698.short?SSO=1')
+)
+
+############################################################################
+try:
+    upload_id = dataset.publish()
+    print('URL for dataset: %s/data/%d' % (openml.config.server, upload_id))
+except openml.exceptions.PyOpenMLError as err:
+    print("OpenML: {0}".format(err))
+
+############################################################################
+# Uploading a dataset stored in a Pandas DataFrame
+############################################################################
+
+############################################################################
+# I might happen that your dataset is made of heterogeneous data which can be
+# usually stored as a Pandas DataFrame. DataFrame offers the adavantages to
+# store the type of data for each column as well as the attribute names.
+# Therefore, when providing a Pandas DataFrame, OpenML can infer those
+# information without the need to specifically provide them when calling the
+# function :func:`create_dataset`. In this regard, you only need to pass
+# ``'auto'`` to the ``attributes`` parameter.
+
+############################################################################
+# Create a fake minimalist dataset stored inside a dataframe.
+
+data = [
+    ['a', 'sunny', 85.0, 85.0, 'FALSE', 'no'],
+    ['b', 'sunny', 80.0, 90.0, 'TRUE', 'no'],
+    ['c', 'overcast', 83.0, 86.0, 'FALSE', 'yes'],
+    ['d', 'rainy', 70.0, 96.0, 'FALSE', 'yes'],
+    ['e', 'rainy', 68.0, 80.0, 'FALSE', 'yes'],
+    ['f', 'rainy', 65.0, 70.0, 'TRUE', 'no'],
+    ['g', 'overcast', 64.0, 65.0, 'TRUE', 'yes'],
+    ['h', 'sunny', 72.0, 95.0, 'FALSE', 'no'],
+    ['i', 'sunny', 69.0, 70.0, 'FALSE', 'yes'],
+    ['j', 'rainy', 75.0, 80.0, 'FALSE', 'yes'],
+    ['k', 'sunny', 75.0, 70.0, 'TRUE', 'yes'],
+    ['l', 'overcast', 72.0, 90.0, 'TRUE', 'yes'],
+    ['m', 'overcast', 81.0, 75.0, 'FALSE', 'yes'],
+    ['n', 'rainy', 71.0, 91.0, 'TRUE', 'no']
+]
+column_names = ['rnd_str', 'outlook', 'temperature', 'humidity',
+                'windy', 'play']
+df = pd.DataFrame(data, columns=column_names)
+# enforce the categorical column to have a categorical dtype
+df['outlook'] = df['outlook'].astype('category')
+df['windy'] = df['windy'].astype('category')
+df['play'] = df['play'].astype('category')
+print(df.info())
+
+############################################################################
+# We enforce the column 'outlook', 'winday', and 'play' to be a categorical
+# dtype while the column 'rnd_str' is kept as a string column. Then, we can
+# call :func:`create_dataset` by passing the dataframe and fixing the parameter
+# ``attributes`` to ``'auto'``.
+
+# force OpenML to infer the attributes from the dataframe
```

Comment:
Is there a reason to define these arguments outside of the function you'll be calling?

Reply:
I found it more didactic to have a comment and an assignment instead of just assigning inside the function. WDYT?

Reply:
The current example has comments inside the function so everything is in one place. Not sure which one is better though as the current version has a pretty long function call.

Reply:
I discussed this with @ArlindKadra regarding #547 to have this based on the length. But that would distribute the creation of the arguments. I start to see why creating them beforehand is a good idea.